akka-http/src/main/scala/akka/http/scaladsl/coding/Gzip.scala

```diff
@@ -4,10 +4,13 @@
 
 package akka.http.scaladsl.coding
 
+import akka.annotation.InternalApi
 import akka.http.scaladsl.model._
 import akka.http.scaladsl.model.headers.HttpEncodings
 
-class Gzip private (compressionLevel: Int, val messageFilter: HttpMessage => Boolean) extends Coder with StreamDecoder {
+@InternalApi
+@deprecated("Actual implementation of Gzip is internal, use Coders.Gzip instead", since = "10.2.0")
```

Comment:
It's a bit strange to mark internal API as deprecated, but indeed as it used to be public API I guess it makes sense :+1:

Reply:
Yeah, guess I went over the top with deprecating stuff, just to be sure. Maybe we can create an internal copy later as the future version while keeping that one around only for compatibility.